src/server/getters/get-markets-awaiting-designated-reporting.ts

```diff
@@ -1,7 +1,24 @@
+import { each } from "async";
 import * as Knex from "knex";
-import { Address } from "../../types";
+import { Address, MarketsRow, UIMarketInfo, UIMarketsInfo, ErrorCallback } from "../../types";
+import { reshapeMarketsRowToUIMarketInfo } from "./get-market-info";
 
-// Look up all markets that are currently awaiting designated (automated) reporting. Should accept an designatedReporterAddress parameter that allows me to filter by designated reporter address.
+// Look up all markets that are currently awaiting designated (automated) reporting.
+// Should accept a designatedReporterAddress parameter that filters by designated reporter address.
 export function getMarketsAwaitingDesignatedReporting(db: Knex, designatedReporter: Address, callback: (err?: Error|null, result?: any) => void): void {
```

Comment:
if `designatedReporter` can be null, specify this in the type `designatedReporter: Addess | null`

Reply:
Unrelated to the above comment, but it caused me to look at this code:

If you control the callsite or can influence its behavior, I recommend either `apple?: type` or `apple: type|null`.  `apple?: type|null` is ultra-permissive which increases the chance of bugs.  In general, if possible I recommend constraining accepted types as much as possible.  (commenting on `err?: Error|null` which is one step away from `any`)

Reply:
I *think* we could generally use `err: Error|null`.  We're using the standard Node-style callbacks, so callbacks are always either `callback(err)` or `callback(null, result)`.  Can built-in knex functions call callbacks with the error argument _missing_?  (@pgebheim would know better than me)

Either way, since we'd be updating the error callback signatures in a ton of different functions, that'd be a separate PR if we want to go that route.